src/CoreOperators.ts

```diff
@@ -15,7 +15,7 @@ export interface CoreOperators<T> {
   combineAll?: <R>(project?: (...values: Array<any>) => R) => Observable<R>;
   combineLatest?: <R>(...observables: Array<Observable<any> | ((...values: Array<any>) => R)>) => Observable<R>;
   concat?: <R>(...observables: (Observable<any> | Scheduler)[]) => Observable<R>;
-  concatAll?: () => Observable<any>;
+  concatAll?: () => Observable<T>;
```

Comment:
Stub code might be like this,
```
Rx.Observable.range(0,3).map(function (x) {
  if (x === 1) {
    return Rx.Observable.range(0,3).mapTo('a');
  }
  return Rx.Observable.range(x, 2);
}).concatAll().subscribe(console.log);
```

output becomes `Observable<number | string>` in this case.
>0
1
a
a
a
2
3